### PR TITLE
chore: rename algo to function

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Description

Rename Algo concept to Function on all the stack.

## Companion PRs

- https://github.com/Substra/orchestrator/pull/139
- https://github.com/Substra/substra-backend/pull/573
- https://github.com/Substra/substra-tests/pull/243
- https://github.com/Substra/substra/pull/341
- https://github.com/Substra/substrafl/pull/82
- https://github.com/Substra/substra-documentation/pull/264
- https://github.com/Substra/substra-frontend/pull/170

### CODE REVIEW

File: examples/titanic_example/run_titanic.py [hunk 1]
```diff
@@ -83,7 +83,7 @@
 #
 # - Data cannot be seen once it's registered on the platform.
 # - Metadata are visible by all the users of a channel.
-# - Permissions allow you to execute an algorithm on a certain dataset.
+# - Permissions allow you to execute an functionrithm on a certain dataset.
```
Comment (RomainGoussault) [inline examples/titanic_example/run_titanic.py:86]:
```suggestion
# - Permissions allow you to execute a function on a certain dataset.
```

File: docs/source/documentation/substra_tools.rst [hunk 2]
```diff
@@ -5,14 +5,14 @@ In Substra, users create tasks that are registered to the platform, then execute
 
 A task needs a valid Dockerfile to create a container and expose a command line interface. The execution of the command creates the expected output files.
 
-For example, an algo defines a list of inputs and outputs. At the task execution, the inputs files are given to the container, the paths to the files are given as arguments to the command line, and the task is responsible for creating the output files.
+For example, an function defines a list of inputs and outputs. At the task execution, the inputs files are given to the container, the paths to the files are given as arguments to the command line, and the task is responsible for creating the output files.
```
Comment (RomainGoussault) [inline docs/source/documentation/substra_tools.rst:8]:
```suggestion
For example, a function defines a list of inputs and outputs. At the task execution, the inputs files are given to the container, the paths to the files are given as arguments to the command line, and the task is responsible for creating the output files.
```

File: examples/titanic_example/run_titanic.py [hunk 3]
```diff
@@ -199,90 +199,90 @@
...
 print(f"Metric key {metric_key}")
 
 
 # %%
-# Adding Algo
+# Adding Function
 # ===========
-# An algorithm specifies the method to train a model on a dataset or the method to aggregate models.
-# Concretely, an algorithm corresponds to an archive (tar or zip file) containing:
+# An functionrithm specifies the method to train a model on a dataset or the method to aggregate models.
```
Comment (RomainGoussault) [inline examples/titanic_example/run_titanic.py:210]:
```suggestion
# A function specifies the method to train a model on a dataset or the method to aggregate models.
```

File: examples/titanic_example/run_titanic.py [hunk 4]
```diff
@@ -199,90 +199,90 @@
...
+# Concretely, an functionrithm corresponds to an archive (tar or zip file) containing:
```
Comment (RomainGoussault) [inline examples/titanic_example/run_titanic.py:211]:
```suggestion
# Concretely, a function corresponds to an archive (tar or zip file) containing:
```

File: examples/titanic_example/run_titanic.py [hunk 5]
```diff
@@ -199,90 +199,90 @@
...
 #
-# - One or more Python scripts that implement the algorithm. Importantly, a train and a
+# - One or more Python scripts that implement the functionrithm. Importantly, a train and a
```
Comment (RomainGoussault) [inline examples/titanic_example/run_titanic.py:213]:
```suggestion
# - One or more Python scripts that implement the function. Importantly, a train and a
```

**Review by RomainGoussault — COMMENTED** [review]
Thanks for the PR!

Why did you add these submodules:
image

**Comment by ThibaultFy** [comment]
> Thanks for the PR!
> 
> Why did you add these submodules: image

Oups that come from the fact that I built the doc modifying the conf.py file... thanks for the catch, and for the review !

**Review by RomainGoussault — APPROVED** [review]
LGTM

**Pull request merged**